### PR TITLE
check if numChanges is null and handle this case

### LINKED ISSUE #316: App crashes when updating trip - NullPointerException in TripDetailActivity.isTheSameTrip()
Version: 1.1.7

When updating a connection / a trip in the TripDetailActivity, the app crashes with the error message "Unfortunately, Transportr has stopped". `adb logcat` gives the following stacktrace:

```
E/AndroidRuntime(27760): FATAL EXCEPTION: main
E/AndroidRuntime(27760): Process: de.grobox.liberario, PID: 27760
E/AndroidRuntime(27760): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Integer.equals(java.lang.Object)' on a null object reference
E/AndroidRuntime(27760): 	at de.grobox.liberario.activities.TripDetailActivity.isTheSameTrip(TripDetailActivity.java:221)
E/AndroidRuntime(27760): 	at de.grobox.liberario.activities.TripDetailActivity.onTripRetrieved(TripDetailActivity.java:152)
E/AndroidRuntime(27760): 	at de.grobox.liberario.tasks.AsyncQueryTripsTask.onPostExecute(AsyncQueryTripsTask.java:152)
E/AndroidRuntime(27760): 	at de.grobox.liberario.tasks.AsyncQueryTripsTask.onPostExecute(AsyncQueryTripsTask.java:42)
E/AndroidRuntime(27760): 	at android.os.AsyncTask.finish(AsyncTask.java:636)
E/AndroidRuntime(27760): 	at android.os.AsyncTask.access$500(AsyncTask.java:177)
E/AndroidRuntime(27760): 	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:653)
E/AndroidRuntime(27760): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(27760): 	at android.os.Looper.loop(Looper.java:135)
...
```

As far as I can remember, this error has been introduced in 1.1.7 and has not occurred before. I hope that the stracktrace helps fixing it.

### PR DESCRIPTION
**Pull request by patkan**
fixes #316 

This adds an additional check in order to not call equals on a null reference.
If the PR is OK I will also cherry-pick this to master.

### CODE REVIEW

**Comment by grote** [comment]
Thanks for the PR, but wouldn't it be better to annotate this `@NonNull` at upstream and ensure it is never `null` there?

**Comment by patkan** [comment]
You're welcome. I have just asked in schildbach/public-transport-enabler#151 if this variable can be not null.

File: src/de/grobox/liberario/activities/TripDetailActivity.java [hunk 1]
```diff
@@ -218,7 +218,13 @@ private boolean isTheSameTrip(Trip old_trip, Trip new_trip) {
 		if(old_trip == null || new_trip == null || new_trip.legs == null) return false;
 
 		// check number of changes and legs
-		if(old_trip.numChanges.equals(new_trip.numChanges) && old_trip.legs.size() == new_trip.legs.size()) {
+		boolean numChangesEq = true;
+		boolean legsEq;
+		if (old_trip.numChanges != null && !old_trip.numChanges.equals(new_trip.numChanges)) {
+			numChangesEq = false;
+		}
+		legsEq = (old_trip.legs.size() == new_trip.legs.size());
```
Comment (grote) [inline src/de/grobox/liberario/activities/TripDetailActivity.java]: Nitpick: You could declare the boolean here

**Comment by grote** [comment]
Thanks, good you please also submit an PR for master, because this code most likely stays unchanged.

**Pull request merged**